app/models/mixins/authentication_mixin.rb

```diff
@@ -341,6 +341,54 @@ def default_authentication
     authentication_type(default_authentication_type)
   end
 
+  # Changes the password of userId on provider client and database.
+  #
+  # @param [current_password] password currently used for connected userId in provider client
+  # @param [new_password]     password that will replace the current one
+  # @param [confirm_password] must confirm the value of the `new_password`
+  #
+  # @return [Boolean] true if the routine is executed successfully
+  #
+  def change_password(current_password, new_password, confirm_password, auth_type = :default)
```

Comment:
I think this looks good - are all ems' `supports_change_password? # => false` by default? Or does that need to be made explicit? Presumably there's an accompanying change in the lenovo provider that turns on support for this?